clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -555,8 +555,19 @@ public synchronized void removeReporter(MetricsReporter reporter) {
 
     synchronized void registerMetric(KafkaMetric metric) {
         MetricName metricName = metric.metricName();
-        if (this.metrics.containsKey(metricName))
-            throw new IllegalArgumentException("A metric named '" + metricName + "' already exists, can't register another one.");
+
+        KafkaMetric existingMetric = this.metrics.get(metricName);
+        if (existingMetric != null) {
```

Comment:
you may want to make this behaviour controllable (at least as a static boolean) so that testing code can continue to explode.

Reply:
That's a pretty good idea. I ended up doing that as you described with a static boolean, since we apparently have several Metrics repositories in the process, and I think we want this to be a process-wide policy.

The default is to not replace on duplicate, so an exception will be thrown in tests.